### PR TITLE
Text attachments are collapsible

### PR DESCRIPTION
**Pull request by Johnlon**
As per discussion https://github.com/cucumber/cucumber-react/issues/78

### 🤔 What's changed?

All text attachments are now collapsible using the HTML "details" tag.

### ⚡️ What's your motivation? 

Long or multiple attachments make the report impossible to read.

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)
- :boom: Breaking change (will cause existing user behaviour to not
  work it did before)

### ♻️ Anything particular you want feedback on?

As per the discussion here https://github.com/cucumber/cucumber-react/issues/78 this changes the behaviour so that the text attachments are collapsed.

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [x] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

### CODE REVIEW

**Comment by Johnlon** [comment]
@mattwynne as requested here's the PR the "npm test" passes.
Not sure where or if to document the rendering change.

File: CHANGELOG.md
Comment (mattwynne) [inline CHANGELOG.md:10]: We tend to just put changes into the `[Unreleased]` section right up until the point where we actually make the release, since we might merge multiple PRs and it will get more awkward to merge this file if they're modifying the version headers.

**Comment by mattwynne** [comment]
Looks like there's some linting to fix.

Could we see a screenshot example in the context of a whole feature file?

**Review by aurelien-reeves — CHANGES_REQUESTED** [review]
Could you please revert the version bump?
We manage that when actually releasing

Also, as @mattwynne already mentioned: could you please fix the linting issues? That would make the review easier

File: src/components/gherkin/Attachment.tsx
Comment (davidjgoss) [inline src/components/gherkin/Attachment.tsx:114]: Where there's no defined fileName, I think it would be useful to append the media type to the fallback message, e.g. `Attached Text (application/json)`.

File: CHANGELOG.md
Comment (davidjgoss) [inline CHANGELOG.md:12]: I think it's worth adding an entry under "Added" that we now display filenames from attachments.

**Review by davidjgoss — COMMENTED** [review]
Thanks for doing this, nice usage of the details/summary pattern and good to display the filenames (which I didn't know was a thing, so we need to update cucumber-js to support these!).

Couple of suggestions, and per other feedback need to roll back the version bump, but looking good.

File: src/components/gherkin/Attachment.tsx
Comment (aurelien-reeves) [reply src/components/gherkin/Attachment.tsx:114]: Also, I guess `??` would be perfectly used here rather than the ternary operator

Comment (davidjgoss) [reply src/components/gherkin/Attachment.tsx:114]: Agreed

File: CHANGELOG.md
Comment (Johnlon) [reply CHANGELOG.md]: done

**Comment by Johnlon** [comment]
All changes made.

@mattwynne  "Could we see a screenshot example in the context of a whole feature file?"

I don't know how to do that. 
i.e I dont know how to prepare a version of cuke that's including this patch so I can't produce the sample report you are looking for.

**Comment by Johnlon** [comment]
I found the story book thing (no documentation for this) showed some of the new features.
But I couldn't figure out how to change the thing to include an "attachment name".

image

**Comment by Johnlon** [comment]
Is there any doco on how to update the sample features and such to demonstrate the new capabilities?

File: CHANGELOG.md
Comment (Johnlon) [reply CHANGELOG.md:10]: done

**Comment by davidjgoss** [comment]
> Is there any doco on how to update the sample features and such to demonstrate the new capabilities?

The test data comes from the [compatibility-kit](https://github.com/cucumber/common/tree/main/compatibility-kit), which we use to validate that different Cucumber implementations behave consistently, and also reuse here for the stories. It would need changes there as well as some in [fake-cucumber](https://github.com/cucumber/common/tree/main/fake-cucumber) (the "reference" impl of Cucumber we use to generate the data). I can give you some more detailed pointers for that, although I don't think it's a blocker for getting this merged, as you've added good tests and I can see it works (having also fiddled with the test data).

File: .eslintrc.json
Comment (davidjgoss) [inline .eslintrc.json:42]: What problem did this solve for you? Also I think if we change this it should be done in `.prettierrc.json`, but could we do that in a different PR?

Comment (Johnlon) [reply .eslintrc.json:42]: This gives a green slate when doing a git clone on Windows where the line endings are different.

Comment (Johnlon) [reply .eslintrc.json]: Removed

Comment (aurelien-reeves) [reply .eslintrc.json]: This would be better managed with git rather than prettier IMO

Comment (davidjgoss) [reply .eslintrc.json]: Yeah I'd forgotten how we do this with cucumber-js and now recall that it's https://github.com/cucumber/cucumber-js/blob/main/.gitattributes#L1

**Review by davidjgoss — APPROVED** [review]
Great change - thanks for taking the time. This should make its way to Cucumber implementations fairly soon.

**Review by aurelien-reeves — APPROVED** [review]

**Comment by aurelien-reeves** [comment]
Thanks @Johnlon 
Great change indeed :D

**Comment by aslakhellesoy** [comment]
Hi @Johnlon,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**
